.husky/pre-commit

```diff
@@ -160,114 +202,201 @@ doesFileContainDotEnvSecret() {
-      echo "$LINE"
-      echo ""
+    local LINE_NUMBER=${LINE%%:*}
+    local LINE_CONTENT=${LINE#*:}
 
+    if is_known_false_positive_match "$STAGED_FILE" "$LINE_NUMBER"; then
+      continue
     fi
 
-  done <<< "$(echo "$MATCHES")"
+    MATCHES+="Potential private key found:\n$DISPLAY_PATH:$LINE_NUMBER:$LINE_CONTENT\n\n"
+  done < <(grep -E -n -- "$ETH_PRIVATE_KEY_PATTERN" "$STAGED_FILE" || true)
 
+  printf "%b" "$MATCHES"
 }
 
-processGitDiff() {
-  echo "-------------------------------------- RESULTS: ---------------------------------------------––"
-  echo ""
+run_secret_checks() {
+  SECRET_FOUND=0
+  PRIVATE_KEY_FOUND=0
+  SECRET_RESULTS=""
 
-  # Check for private keys and secrets in all added or modified FILES
-  git diff --cached --name-only --diff-filter=ACM | while IFS= read -r FILE; do
-    # Skip excluded paths
-    for EXCLUDED_PATH in "${EXCLUDED_PATHS[@]}"; do
-      if [[ "$FILE" == "$EXCLUDED_PATH"* ]]; then
-        continue 2
-      fi
-    done
+  while IFS= read -r -d '' FILE; do
+    if is_excluded_path "$FILE"; then
+      continue
+    fi
+
+    local STAGED_COPY
+    STAGED_COPY=$(mktemp "$TEMP_DIR/staged.XXXXXX")
 
-    # Check for secrets from .env file
-    RESULT_SECRET=$(doesFileContainDotEnvSecret "$FILE")
-    if [[ -n $RESULT_SECRET ]]; then
-      printRed "$RESULT_SECRET"
-      echo ""
+    if ! git show ":$FILE" > "$STAGED_COPY" 2>/dev/null; then
+      rm -f "$STAGED_COPY"
+      continue
     fi
 
-    # Check for potential private keys
-    RESULT_PRIVKEY=$(doesFileContainPotentialPrivateKey "$FILE")
-    if [[ -n $RESULT_PRIVKEY ]]; then
-      printYellow "$RESULT_PRIVKEY"
-      echo ""
+    local RESULT_SECRET
+    RESULT_SECRET=$(does_file_contain_dot_env_secret "$STAGED_COPY" "$FILE")
+    if [ -n "$RESULT_SECRET" ]; then
+      SECRET_RESULTS+="$RESULT_SECRET\n"
+      SECRET_FOUND=1
     fi
-  done
-  echo "---------------------------------------------------------------------------------------------––"
+
+    local RESULT_PRIVKEY
+    RESULT_PRIVKEY=$(does_file_contain_potential_private_key "$STAGED_COPY" "$FILE")
+    if [ -n "$RESULT_PRIVKEY" ]; then
+      SECRET_RESULTS+="$RESULT_PRIVKEY\n"
+      PRIVATE_KEY_FOUND=1
+    fi
+
+    rm -f "$STAGED_COPY"
+  done < <(git diff --cached --name-only --diff-filter=ACM -z)
 }
 
-checkGitDiffForSecretsAndPrivateKeys() {
+prepare_env_secrets
+
+# Start forge build and lint-staged in parallel (they're independent initially)
+FORGE_PID=""
+LINT_STAGED_PID=""
+
+print_section "Pre-commit Checks"
+
+# Start parallel tasks
+if [ "$HAS_SOL_FILES" = "yes" ]; then
+  print_status "info" "Building contracts with forge..."
+  (forge build --skip test > "$FORGE_OUTPUT" 2>&1; echo $? > "$FORGE_EXIT") &
+  FORGE_PID=$!
+else
+  print_status "skip" "Skipping forge build (no Solidity files changed)"
+  echo 0 > "$FORGE_EXIT"
+fi
 
-  # process all files in git diff and search for secrets
-  local RESULT=$(processGitDiff)
+if [ "$HAS_TYPE_FILES" = "yes" ]; then
```

Comment:
I’m not sure we gain much from this and it adds extra complexity. lint-staged is already scoped in package.json to the file extensions it should scan. If we introduce new linting rules for additional extensions, we’d have to remember to update both package.json and the pre-commit script